.buildkite/pipeline.full.yml

```diff
@@ -268,12 +346,198 @@ steps:
       docker-compose#v3.7.0:
         run: android-maze-runner
         command:
-          - "features/full_tests"
+          - "features/full_tests/batch_1"
+          - "--app=/app/build/release/fixture-r21.apk"
+          - "--farm=bs"
+          - "--device=ANDROID_11_0"
+          - "--fail-fast"
+    concurrency: 9
+    concurrency_group: 'browserstack-app'
+
+  - label: ':android: Android 11 NDK r21 end-to-end tests - batch 2'
+    depends_on:
+      - "fixture-r21"
+      - "android-11-smoke"
+    timeout_in_minutes: 60
+    plugins:
+      artifacts#v1.2.0:
+        download: "build/release/fixture-r21.apk"
+      docker-compose#v3.7.0:
+        run: android-maze-runner
+        command:
+          - "features/full_tests/batch_2"
           - "--app=/app/build/release/fixture-r21.apk"
           - "--farm=bs"
           - "--device=ANDROID_11_0"
-          - "--username=$BROWSER_STACK_USERNAME"
-          - "--access-key=$BROWSER_STACK_ACCESS_KEY"
           - "--fail-fast"
     concurrency: 9
     concurrency_group: 'browserstack-app'
+
+  #
+  # Flaky features
+  #
+  - label: ':hankey: Flaky Android 4.4 NDK r16 end-to-end tests'
+    depends_on:
+      - "fixture-r16"
+    timeout_in_minutes: 90
+    plugins:
+      artifacts#v1.2.0:
+        download: "build/release/fixture-r16.apk"
+      docker-compose#v3.7.0:
+        run: android-maze-runner
+        command:
+          - "features/full_tests/flaky.feature"
+          - "--retry=2"
+          - "--strict-undefined"
+          - "--strict-pending"
+          - "--app=/app/build/release/fixture-r16.apk"
+          - "--farm=bs"
+          - "--device=ANDROID_4_4"
+    concurrency: 9
+    concurrency_group: 'browserstack-app'
+
+  - label: ':hankey: Flaky Android 5 NDK r16 end-to-end tests'
```

Comment:
Probably a future enhancement, but is there any way to reduce the length of these yaml files? A lot of the jobs are doing the same thing but only altering the fixture APK/device, so perhaps there's a case for generating the yaml from some sort of script that has a test matrix?

Reply:
We've discussed a bit recently and I think @imjoehaines may be trialling an approach on our Ruby notifier.  Certainly one to keep in mind and consideration once the pipeline has settled down fully.